python/lsst/scarlet/lite/component.py

```diff
@@ -265,7 +266,11 @@ def prox_morph(self, morph: np.ndarray) -> np.ndarray:
             )
         morph[peak] = np.max([morph[peak], self.floor])
         # Normalize the morphology
-        morph[:] = morph / np.max(morph)
+        maxValue = np.max(morph)
```

Comment:
What was the issue before? Negative values in `morph`, or `NaN` entries? If the latter, would `np.nanmax` make more sense here?

Reply:
I understand your thinking, but a NaN is an indication that something went wrong, so better to set it to 0 where it won't harm the other models. But it's actually the division by zero that is most likely to have caused the warning that you were seeing.

Reply:
If `morph` has any `NaN` values in it, then `maxValue = np.nan(morph)` will return `nan`, so no division of `morph` will take place. Line 273 below only sets `morph` elements to zero where `morph` is not finite - i.e., the non-NaN entires in `morph` will be returned but will not have been divided by any max value. Is that ok?

Reply:
Ah, I see. You're right, I missed that. Thanks for the catch.

Reply:
I actually did one better, because this could still run into problems if there is an infinite value. So I moved the `isfinite` check, which excludes infinite and NaN values, to before the maximum value is obtained.

Reply:
Looks good to me!